alphagov/paas-cf
# [#111676128] Auto generate ssl cert for deployer concourse

# What
Story: https://www.pivotaltracker.com/n/projects/1275640
We generate a self signed certificate for each environment and set up the Concourse ELB with it. It is based on the environment name and the terraform variable `dns_zone_name`.

The certificate is stored in S3. If it is already present in S3, then it will use this one. This will be useful for manually generated certificates.

# How to review
* Create a new deployer Concourse
* Check it uses a newly generated certificate.
* Check if you run the second time the certificate is not changed
* Check the certificate is removed from IAM when you destroy the environment

# Note
We have observed a bug that looks like a race condition when we tried to delete the certificate from S3 to force the generation. Sometimes Concourse uses the cached resource and will re-upload it.

The workaround is to wait at least 5 min between removing the cert and running the pipeline again.

# Who can review
Anyone but @saliceti or @timmow
